### ISSUE TITLE
Agenda items not preserved when two scrapers collect same event

### ISSUE THREAD

**Issue by patcon** [opencivicdata/pupa#239]
Reticketed from https://github.com/opencivicdata/scrapers-us-municipal/issues/111

---

I have one scraper that scrapes all events scheduled in the legislative session, and another that runs for collecting details about recently passed and upcoming events (for which agendas are just being published).

So essentially:
1. **`events-incremental`.** quick nightly scrape that builds events with agendas around the current date, and
2. **`events-full`.** another scraper for the full schedule, that doesn't touch agendas.

I am seeing that if `events-full` runs after `events-incremental` it blows away all the agendas.

A [slack conversation](https://opencivicdata.slack.com/archives/pupa/p1463506105000003) with @paultag lead me to believe that this shouldn't be happening if no agenda items are added or touched.

However, I've just built a simple scraper a reproduced the wiping behaviour:
https://github.com/patcon/scrapers-us-municipal/tree/agenda-wipe-bug-demo

If I run the below code, the agenda items appear, and then disappear after running the second scrape:

```
pupa update test events_has_agenda
pupa update test events_no_agenda
```

Can someone confirm that this is unexpected behaviour, and perhaps suggest any ideas on where the regression may have happened? Thanks! :)

**Comment by jpmckinney** [comment]
I think this is dup #169

**Comment by patcon** [comment]
I don't think they're dups james. This one is suggesting unexpected behavior where rows from `opencivicdata_eventagendaitem`s are removed from events when a second scrape (with no agenda item data) is run.

The linked issue is about seems to be about a larger issue of no doing more scraping work than required. I don't think it's solution would resolve this issue's concern.

**Comment by jpmckinney** [comment]
Hmm, perhaps `import_item` is what needs to change to resolve this issue.

**Comment by jpmckinney** [comment]
Actually, re-read the other issue, and it seems relevant, like [this comment](https://github.com/opencivicdata/pupa/issues/169#issuecomment-94900817).

The importer can't determine whether the scrape with no agenda items lacks them because it's an intentionally incomplete object, or because those agenda items were removed in the real world.

paultag's comments were too brief to read much into that Slack conversation. It seems to me that this is behaving as designed. I think the resolution would be for you to adjust Pupa's logic for your special case.

Basically, you'd have to skip [this logic](https://github.com/opencivicdata/pupa/blob/226e57435529d6184bbc6209d3cb243ab0773838/pupa/importers/base.py#L254-L255) if the data for the related model (in this case `agenda`) is empty.

I suppose a generic solution would be to have a flag on the `data` object to indicate which `related_models` should have the default 'clear everything' behavior if empty, and which should have the 'do nothing' behavior that you desire. I'm not sure there's appetite to maintain that additional complexity.

**Comment by patcon** [comment]
Ah ok, will admit that had optimism after paltag convo that this was a "simple" regression and the feature already existed, but i realize from that comment you linked that you're totally correct. Thanks for humouring me james!
